### PR TITLE
Minor: Fix bugs in docs for `to_timestamp`, `to_timestamp_seconds`, ...

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

N/A

## Rationale for this change

The docs implied that all of these functions took nanosecond values as input, when in fact they take different precisions. This caused some confusion internally at InfluxData

## What changes are included in this PR?
1. Update the docs to show the correct precision
2. Clarify what types are arguments are handled

## Are these changes tested?
N/A

## Are there any user-facing changes?
Updated documentation

### CODE REVIEW

File: docs/source/user-guide/sql/scalar_functions.md [hunk 1]
```diff
@@ -1407,10 +1408,12 @@ to_timestamp(expression)
 
 ### `to_timestamp_millis`
 
-Converts a value to RFC3339 millisecond timestamp format (`YYYY-MM-DDT00:00:00.000Z`).
-Supports timestamp, integer, and unsigned integer types as input.
-Integers and unsigned integers are parsed as Unix nanosecond timestamps and
```
Comment (alamb) [inline docs/source/user-guide/sql/scalar_functions.md:1412]: this text incorrectly implies numeric arguments are interpreted as `nanoseconds` which is not the case (they are treated as `milliseconds). 

I also tried to clean up the language about what a "unix timetamp" meant (really means seconds from unix epoch)

**Comment by alamb** [comment]
cc cc @Jayclifford345 and @mhilton

File: docs/source/user-guide/sql/scalar_functions.md [hunk 2]
```diff
@@ -1391,10 +1391,11 @@ extract(field FROM source)
 
 ### `to_timestamp`
```
Comment (comphead) [inline docs/source/user-guide/sql/scalar_functions.md:1393]: 👍

File: docs/source/user-guide/sql/scalar_functions.md [hunk 3]
```diff
@@ -1423,21 +1425,23 @@ to_timestamp_millis(expression)
 
 ### `to_timestamp_micros`
 
-Converts a value to RFC3339 microsecond timestamp format (`YYYY-MM-DDT00:00:00.000000Z`).
-Supports timestamp, integer, and unsigned integer types as input.
-Integers and unsigned integers are parsed as Unix nanosecond timestamps and
-return the corresponding RFC3339 timestamp.
+Converts a value to a timestamp (`YYYY-MM-DDT00:00:00.000Z`).
```
Comment (comphead) [inline docs/source/user-guide/sql/scalar_functions.md:1428]:
```suggestion
Converts a value to a timestamp (`YYYY-MM-DDT00:00:00.000000Z`).
```

File: docs/source/user-guide/sql/scalar_functions.md [hunk 4]
```diff
@@ -1423,21 +1425,23 @@ to_timestamp_millis(expression)
...
 to_timestamp_nanos(expression)
 ```
 
 ### `to_timestamp_nanos`
 
-Converts a value to RFC3339 nanosecond timestamp format (`YYYY-MM-DDT00:00:00.000000000Z`).
-Supports timestamp, integer, and unsigned integer types as input.
-Integers and unsigned integers are parsed as Unix nanosecond timestamps and
-return the corresponding RFC3339 timestamp.
+Converts a value to a timestamp (`YYYY-MM-DDT00:00:00.000Z`).
```
Comment (comphead) [inline docs/source/user-guide/sql/scalar_functions.md:1440]:
```suggestion
Converts a value to a timestamp (`YYYY-MM-DDT00:00:00.000000000Z`).
```

**Review by comphead — COMMENTED** [review]
Thanks @alamb some suggestions

File: docs/source/user-guide/sql/scalar_functions.md (hunk 3 above)
Comment (comphead) [reply docs/source/user-guide/sql/scalar_functions.md:1428]: This suggestion would be nice to commit as well

**Review by comphead — APPROVED** [review]
lgtm, thanks @alamb

**Pull request merged**